enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/Reference.java

```diff
@@ -216,16 +197,21 @@ private static int maxBits()
 
     public static <SOURCE> long decode( SOURCE source, DataAdapter<SOURCE> adapter )
     {
-        short firstByte = (short) (adapter.get( source ) & 0xFF);
-        for ( Reference encoding : ENCODINGS )
+        int header = adapter.getByte( source ) & 0xFF;
```

Comment:
For maintainability it would be brilliant with comments on most of these lines describing roughly what happens.